### PR TITLE
chore(Tactic/Lint/Simp): make simpNF linter dsimp-aware

### PR DESCRIPTION
**Pull request by jcommelin**

### CODE REVIEW

File: Batteries/Tactic/Lint/Simp.lean [hunk 1]
```diff
@@ -112,11 +112,18 @@ https://leanprover-community.github.io/mathlib_docs/notes.html#simp-normal%20for
     unless ← isSimpTheorem declName do return none
     let ctx := { ← Simp.Context.mkDefault with config.decide := false }
     checkAllSimpTheoremInfos (← getConstInfo declName).type fun {lhs, rhs, isConditional, ..} => do
+      let isRfl ← isRflTheorem declName
       let ({ expr := lhs', proof? := prf1, .. }, prf1Stats) ←
-        decorateError "simplify fails on left-hand side:" <| simp lhs ctx
+        decorateError "simplify fails on left-hand side:" <|
+          if !isRfl then simp lhs ctx else do
+            let (e, s) ← dsimp lhs ctx
+            return (Simp.Result.mk e .none .true, s)
```
Comment (semorrison) [inline Batteries/Tactic/Lint/Simp.lean:120]:
```suggestion
          if !isRfl then 
            simp lhs ctx
          else do
            let (e, s) ← dsimp lhs ctx
            return (Simp.Result.mk e .none .true, s)
```

Comment (semorrison) [reply Batteries/Tactic/Lint/Simp.lean:120]: and below.

**Review by YaelDillies — COMMENTED** [review]
Can you update the error message as well? It currently reads `"simp can prove this:`.

File: Batteries/Tactic/Lint/Simp.lean (hunk 1 above)
Comment (jcommelin) [reply Batteries/Tactic/Lint/Simp.lean]: Done

**Comment by jcommelin** [comment]
I updated the error message.

**Comment by joneugster** [comment]
> Can you update the error message as well? It currently reads `"simp can prove this:`.

(and also in `formatLemmas` where it says `simp only`: [here](https://github.com/leanprover-community/batteries/blob/a7bf196220562dbe18e3ad5bfcee7cc20789e0e9/Batteries/Tactic/Lint/Simp.lean#L103))

**Review by YaelDillies — APPROVED** [review]
Great, thanks! LGTM

**Review by joneugster — APPROVED** [review]

**Pull request merged**